### PR TITLE
Fix ListItem css

### PR DESCRIPTION
**Pull request by jotoeri**
This fixes two things:
- The ListItem was not really usable without subtitle but with counter. Different line heights and a completely messed up design were the result. This now brings that case onto one line, when only title and counter are available and fixes the ListItem height, so all ListItems are the same height. 

| Before | After |
|--|--|
[Full odd looking gif](https://user-images.githubusercontent.com/47433654/164996305-9cd5c448-f1bd-45de-8328-f6dd66b8ab91.gif)||
|old|non_compact|
 
- In connection to the fixed height, it introduces a compact-mode for the list item. Once used without subtitle or in a mixed case, the default sizes are quite large. Esp. if there are multiple elements without subtitle, but a few with subtitle, there is just too much space inbetween those items without. Thus the compact mode reduces the size of the ListItem a bit, so the mixed case looks good.

| Default | Compact Mode |
|--|--|
|non_compact|compact_mode|

### CODE REVIEW

**Comment by raimund-schluessler** [comment]
@jotoeri Could you please show before and after pictures and describe a bit more what was/is the problem you are fixing here? If we change the padding of `ListItem` and of the `Actions` inside, we might break the alignment with other components in the library, so there should b good reasoning, as we likely need to change it for other components as well then.

**Comment by jotoeri** [comment]
> @jotoeri Could you please show before and after pictures and describe a bit more what was/is the problem you are fixing here? If we change the padding of `ListItem` and of the `Actions` inside, we might break the alignment with other components in the library, so there should b good reasoning, as we likely need to change it for other components as well then.

Well, that's why it has the developing label and i had to check if this is, what i wanted. :wink: 
- You're right, the padding of the list-item is not necessary anymore. Compared to my initial base on the last release, the padding of the appnavigation already increased, so the padding here would be doubled.
- For the actions margin-left i have a specific before/after here. It just increases the space a little bit, if the text overflow gets hidden:

| Before | After |
|---|---|
| actions_old | actions_new2 |
- For the whole PR i'll update the initial comment later :slightly_smiling_face:

**Comment by jotoeri** [comment]
As i still need to double-check these changes with other apps, @raimund-schluessler do you know some apps that use the listItem?

**Comment by Chartman123** [comment]
@jotoeri A quick search within the Nextcloud organisation here on GitHub only brought up the mail app:

https://github.com/search?q=org%3Anextcloud+%22import+ListItem%22&type=code

But there might be of course other apps that are not connected to the organisation here.

**Comment by jotoeri** [comment]
Just checked it on Talk and didn't see any problems. So should be good to go now. :slightly_smiling_face:

File: src/components/ListItem/ListItem.vue [hunk 1]
```diff
@@ -510,16 +534,19 @@ export default {
...
+				display: flex;
+			}
 		}
 
 		&__actions {
 			flex: 0 0 auto;
 			align-self: center;
 			justify-content: center;
-
+			margin-left: 4px;
```
Comment (jotoeri) [inline src/components/ListItem/ListItem.vue:549]: | Before | After |
|---|---|
| actions_old | actions_new2 |

File: src/components/ListItem/ListItem.vue [hunk 2]
```diff
@@ -565,15 +593,14 @@ export default {
 	&__subtitle {
 		overflow: hidden;
 		flex-grow: 1;
-		padding-right: 4px;
```
Comment (jotoeri) [inline src/components/ListItem/ListItem.vue:568]: Shifted to counter margin, so it also applies if no subtitle used.

**Review by marcoambrosini — APPROVED** [review]

**Review by Chartman123 — APPROVED** [review]

**Review by nimishavijay — APPROVED** [review]

**Review by marcoambrosini — CHANGES_REQUESTED** [review]
@jotoeri could you add a compact example in the docs too?

**Comment by jotoeri** [comment]
:+1: @marcoambrosini Can i easily get that to half width? Just add a style to the surrounding `ul`? :thinking:
preview

**Comment by marcoambrosini** [comment]
@jotoeri sure thingy :)

**Comment by jotoeri** [comment]
preview

**Review by marcoambrosini — APPROVED** [review]

**Pull request merged**